tests/models/test_transformers.py

```diff
@@ -1,18 +1,59 @@
-import guidance
-from guidance import gen, capture
+from textwrap import dedent
+
+import pytest
+
+from guidance import gen, select
+
 from ..utils import get_model
 
+
 def test_gpt2():
     gpt2 = get_model("transformers:gpt2")
     lm = gpt2 + "this is a test" + gen("test", max_tokens=10)
     assert len(str(lm)) > len("this is a test")
 
+
 def test_recursion_error():
-    '''This checks for an infinite recursion error resulting from a terminal node at the root of a grammar.'''
+    """This checks for an infinite recursion error resulting from a terminal node at the root of a grammar."""
     gpt2 = get_model("transformers:gpt2")
 
     # define a guidance program that adapts a proverb
-    lm = gpt2 + f"""Tweak this proverb to apply to model instructions instead.
+    lm = (
+        gpt2
+        + f"""Tweak this proverb to apply to model instructions instead.
     {gen('verse', max_tokens=2)}
     """
-    assert len(str(lm)) > len("Tweak this proverb to apply to model instructions instead.\n\n")
\ No newline at end of file
+    )
+    assert len(str(lm)) > len(
+        "Tweak this proverb to apply to model instructions instead.\n\n"
+    )
+
+
+TRANSFORMER_MODELS = ["gpt2", "microsoft/phi-2"]
+
+
+@pytest.mark.parametrize("model_name", TRANSFORMER_MODELS)
+def test_transformer_smoke_gen(model_name):
+    my_model = get_model(f"transformers:{model_name}", trust_remote_code=True)
+
+    prompt = dedent(f"""How many sides has a triangle?""")
+    lm = my_model + prompt + gen(name="answer", max_tokens=2)
+    assert len(lm["answer"]) > 0, f"Output: {lm['answer']}"
+    # Inexact, but at least make sure not too much was produced
+    assert len(lm["answer"]) < 8, f"Output: {lm['answer']}"
+
+
+@pytest.mark.parametrize("model_name", TRANSFORMER_MODELS)
+def test_transformer_smoke_select(model_name):
+    my_model = get_model(f"transformers:{model_name}", trust_remote_code=True)
+
+    prompt = dedent(
+        """How many sides has a triangle?
+
+p) 4
+t) 3
+w) 10
+"""
```

Comment:
small nit, but if we're dedenting, we can probably make this fstring look much nicer (or alternatively, just drop the dedent) call

Reply:
Done